libraries/postgres/src/util.js

```diff
@@ -76,3 +76,37 @@ exports.annotateError = (query, err) => {
     }
   }
 };
+
+/**
+ * Call the given fetch method with a given size and offset to fetch data, and
+ * return an async iterator that will yield each row in turn.
+ */
+exports.paginatedIterator = ({fetch, size = 1000}) => {
+  return {
+    [Symbol.asyncIterator]() {
+      let offset = 0;
+      let done = false;
+      let rows = [];
+
+      const next = async () => {
+        if (rows.length > 0) {
+          return {value: rows.shift(), done: false};
+        }
+        if (done) {
+          return {done};
```

Comment:
It looks like we return `{ done: true }` `if rows.length === 0`. Does L96 ever run?

Reply:
I think it's allowed to call `iter.next()` after it has returned `done`, so this case saves an additional DB fetch in that situation.

Reply:
Got it. A comment there would be helpful for posterity.

Reply:
done, and I'll push with that change.